### DISCUSSION TITLE
Why is Prisma generating a @defaulted String for text()

### DISCUSSION THREAD

**Discussion by chirdeeptomar** [keystonejs/keystone#7455] [Questions] [answered]
Playing with Keystone and found a few oddities  with what is generated.  Looking at the default generated application.

Post List has the following field

`      title: text(),
`

Prisma output:

`  title                String    @default("")
`

`@default("") ` is not what is expected, this should be a mandatory coulmn in the db which should raise a db exception if not provided. As a consequence, in graphql Title should be String! which it is not. 

Also, prisma allows you to constrain the cloumn size, no way to restrict at the db as maxlength is not propogated down to the generated prisma schema.

Thanks

**Comment by bladey** [comment]
Thanks @chirdeeptomar, let me check in with the team on this and I'll get back to you.

**Comment by dcousens** [comment, marked as answer]
Hi @chirdeeptomar,

Please [read our technical discussion](https://github.com/keystonejs/keystone/discussions/7158) about why `text()` is defaulted to `""` instead of `null`.
(https://github.com/keystonejs/keystone/discussions/7158)
